Pokedex/Pokedex/Models/Model.swift

```diff
@@ -0,0 +1,74 @@
+//  Copyright © 2019 Frulwinn. All rights reserved.
+
+import Foundation
+
+class Model {
+    
+    static let shared = Model()
+    private init () {}
+    
+    //baseURL for api
+    let baseURL = URL(string: "https://pokeapi.co/api/v2/pokemon/")!
+    
+    //MARK: Properties
+    var pokemon: Pokemon?
+    var pokemons: [Pokemon] = []
+    
+    //MARK: Methods
+    //count
+    func numberOfPokemons() -> Int {
+        return pokemons.count
+    }
+    
+    //access pokemon at index
+    func pokemon(at indexPath: IndexPath) -> Pokemon {
+        return pokemons[indexPath.row]
+    }
+    
+    //add
+    func addPokemon(pokemon: Pokemon) {
+        pokemons.append(pokemon)
+    }
+    
+    //delete
+    func deletePokemon(at indexPath: IndexPath) {
+        pokemons.remove(at: indexPath.row)
+    }
+    
+    //update
+    func updatePokemon(at indexPath: IndexPath) {
+        let _ = pokemons[indexPath.row]
```

Comment:
So even you haven't used it in this project but this function need some modification to work. For updating an object you also need new values to pass into function as argument and then use them to update the object's related values. So if you want to update image of a pokemon you need to pass current pokemon or it's index and new image. Then find that pokemon inside of pokemons array and change it's image